### ISSUE TITLE
FFmpeg Encoding

### ISSUE THREAD

**Issue by acidhax** [Microsoft/FFmpegInterop#9]
Are there functions available to support the file encoding system of FFmpeg?

**Comment by mcosmin222** [comment]
The project itself doesn't have it but you can wrap them the same way they wrapped the decoders.

**Comment by yodurr** [comment]
We haven't focused on the encoding side.  How important is it for you to encode using ffmpeg?

**Comment by acidhax** [comment]
Well, in Videostream, I hit 1.4m users with ffmpeg encoding in the browser. This would be good for media streaming apps in Windows. AppleTV, Roku, Chromecast, etc.. all have a subset of supported video formats.

It would be a hugely appreciated feature!

**Comment by brettabalmer** [comment]
I agree - support for encoding and filters would be an immensely helpful feature.

**Comment by krishnan-unni** [comment]
Yeah me too agree, would be a killer feature if this could have h264 h265 encoders along with rtmp write(upload function). I have written a winrt universal app component  which captures the screen and tries to upload it to a rtmp server(live streaming), I can share it if anyone is interested. But ffmpeg instruction given here does not include libx264 in it so the com is not finding the encoder. I am not able to compile ffmpeg with libx264. The msys is saying cant recognize the system when I try to compile x264 first as per the instruction given in ffmpeg.org.

**Comment by bobybacs** [comment]
@krishnan-unni: I'm interested about that winrt universal app component, could you please share it ?
Thanks.

**Comment by krishnan-unni** [comment]
@bobybacs I'll upload it and send the link. But keep this in mind, I have not been able to test the solution since I couldn't get the libx264 compiled with ffmpeg for windows10 x64. So the av_codec_find_encoder function is not able to find the H264 encoder. And its available for video only now not audio. There could be other errors, its still a WIP code.

**Comment by qzzf1w** [comment]
I think it is very important. Screen recording is a huge gap in UWP app. And a key shortcoming of surface hub for MS.

**Comment by ottolb** [comment]
Hi,
@krishnan-unni can you send me the link to your screen capture project? 
Thanks

**Comment by valentin-dan** [comment]
Hello,

Mostly I need a way to encode so I'll post this here:

1. I'd like to save a RTSP to file while also offering a preview. The preview part is done in the sample; how can I save the stream ?

2. RTSP streamming from VLC connected to my camera works just fine if captured by another VLC over the internet, but works very slow if i capture it with the sample project; any clue as to why ? (VLC is set to H.264 + MP3(Mp4) RTSP)

Thanks !

**Comment by Liero** [comment]
I need to reencode WAV as OGG files. This would be highly appreciated.
